types/deep-diff/index.d.ts

```diff
@@ -1,42 +1,60 @@
-// Type definitions for deep-diff
+// Type definitions for deep-diff 1.0
 // Project: https://github.com/flitbit/diff/
 // Definitions by: ZauberNerd <https://github.com/ZauberNerd>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 2.3
 
-declare namespace deepDiff {
-    interface IDiff {
-        kind: string;
-        path: string[];
-        lhs: any;
-        rhs: any;
-        index?: number;
-        item?: IDiff;
-    }
-
-    interface IAccumulator {
-        push(diff: IDiff): void;
-        length: number;
-    }
-
-    interface IPrefilter {
-        (path: string[], key: string): boolean;
-    }
-
-    interface IDeepDiff {        
-        diff(lhs: Object, rhs: Object, prefilter?: IPrefilter, acc?: IAccumulator): IDiff[];
-        diff(): IDiff;
-        observableDiff(lhs: Object, rhs: Object, changes: Function, prefilter?: IPrefilter, path?: string[], key?: string, stack?: Object[]): void;
-        applyDiff(target: Object, source: Object, filter: Function): void;
-        applyChange(target: Object, source: Object, change: IDiff): void;
-        revertChange(target: Object, source: Object, change: IDiff): void;
-        isConflict(): boolean;
-        noConflict(): IDeepDiff;
-    }
+export interface DiffNew<RHS, Kind = 'N'> {
```

Comment:
Why do you need the second type parameter? It's never given a type argument.

Reply:
In my experience, putting the Kind up there makes type assertions automatic since it makes it clear to the TypeScript type checker that a Diff with Kind = 'N' *has* to be a DiffNew.
This could have changed since I learned about that, though?